File: AdvantageFramework.Core/Database/Entities/AlertType.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AdvantageFramework.Core.Database.Entities
{
    [Table("ALERT_TYPE")]
    public partial class AlertType
    {
        public AlertType()
        {
            AlertCategories = new HashSet<AlertCategory>();
            Alerts = new HashSet<Alert>();
        }

        #region Constants



        #endregion

        #region Enum

        public enum Properties
        {
            TBD
        }


        #endregion

        #region Variables



        #endregion

        #region Properties

        [Key]
        [Column("ALERT_TYPE_ID")]
        public int AlertTypeId { get; set; }
        [Required]
        [Column("ALERT_TYPE_DESC")]
        [StringLength(40)]
        public string AlertTypeDesc { get; set; }

        [InverseProperty(nameof(AlertCategory.AlertType))]
        public virtual ICollection<AlertCategory> AlertCategories { get; set; }
        [InverseProperty(nameof(Alert.AlertType))]
        public virtual ICollection<Alert> Alerts { get; set; }

        #endregion

        #region Methods



        #endregion
    }
}

File: AdvantageFramework/Database/Entities/AlertType.vb
Namespace Database.Entities

	<Table("ALERT_TYPE")>
	Public Class AlertType
		Inherits BaseClasses.Entity

#Region " Constants "



#End Region

#Region " Enum "

		Public Enum Properties
			ID
			Description
			Alerts
			AlertCategories

		End Enum

#End Region

#Region " Variables "



#End Region

#Region " Properties "

		<Key>
		<Required>
		<Column("ALERT_TYPE_ID")>
		Public Property ID() As Integer
		<Required>
		<MaxLength(40)>
		<Column("ALERT_TYPE_DESC", TypeName:="varchar")>
		Public Property Description() As String

        Public Overridable Property Alerts As ICollection(Of Database.Entities.Alert)
        Public Overridable Property AlertCategories As ICollection(Of Database.Entities.AlertCategory)

#End Region

#Region " Methods "

        Public Overrides Function ToString() As String

            ToString = Me.ID

        End Function

#End Region

	End Class

End Namespace
